settings.gradle.kts

```diff
@@ -1,6 +1,9 @@
 buildscript { dependencies { classpath("com.diffplug.spotless:spotless-lib-extra:2.43.1") } }
 
-plugins { id("com.diffplug.configuration-cache-for-platform-specific-build") version "3.44.0" }
+plugins {
+  id("com.diffplug.configuration-cache-for-platform-specific-build") version "3.44.0"
+  id("org.gradle.toolchains.foojay-resolver-convention") version "0.7.0"
```

Comment:
I was a bit hesitant to add this in on another project, since it downloads JDKs and then runs them.  But I guess since it comes from Gradle, it's probably pretty trustworthy?  The alternative to using this would just be documenting which JDK versions need to be installed and installing them ourselves on CI.

Reply:
I think it's trustworthy. It's also widely used, so I'd expect any vulnerability to be discovered quickly. Certainly, this is far from being the only code we download from the Internet and run as part of building and testing WALA.

> The alternative to using this would just be documenting which JDK versions need to be installed and installing them ourselves on CI.

Agreed. But I prefer an automated mechanism over prose instructions that each new WALA developer must manually apply.

That being said, manual pre-installation of suitable JDKs is still an option. If someone is working on WALA in a restricted environment that would prevent [toolchain auto-provisioning](https://docs.gradle.org/current/userguide/toolchains.html#sec:provisioning) from working, that's fine. WALA's Eclipse components will build using whatever JDK 17 [Gradle can find already installed on the system](https://docs.gradle.org/current/userguide/toolchains.html#sec:auto_detection) or [custom-configured](https://docs.gradle.org/current/userguide/toolchains.html#sec:custom_loc).